tests/countries/test_finland.py

```diff
@@ -165,6 +168,163 @@ def test_all_saints_day(self):
             "1957-11-01",
         )
 
+    def _test_unofficial_holiday(self, name, date_first_observed_on):
+        year = date_first_observed_on.year
+        month = date_first_observed_on.month
+        day = date_first_observed_on.day
```

Comment:
```suggestion
        year, month, day = map(int, date_first_observed_on.split("-"))
```
I think it would be more optimal not to use `datetime`, but just a string with date in ISO format.

Reply:
Hm, well, I could do that as well. It's just that I anyway need to parse out year from passed string, and though of doing that with [strptime](https://docs.python.org/3/library/datetime.html#datetime.datetime.strptime), at which point it felt easier to just pass date object as is.

But I can of course just pass `yyyy-mm-dd` string and split it by `-` and assume that year, month and date are on the right positions.

Reply:
You're right -- it'll need to be parsed anyway. It's just more readable and performant (vs datetime objects) to have parameters as strings and do the parsing within the `_test_unofficial_holiday()` either via map or date (you don't need the actual date object, right? so @KJhellico's approach looks more efficient)

// btw, `datetime.date` instead of `datetime.datetime` would be enough

> Out of curiosity, why not use datetime in tests still? Feels like it would be a bit more robust way from accidental mishaps with string formatting and validation.

I think I refactored tests to look this way for better readability. We use datetime.* where appropriate of course.

Reply:
It seems it started [here](https://github.com/vacanza/python-holidays/pull/800).

Reply:
> Out of curiosity, why not use datetime in tests still?

Somehow, historically, it happened that way. :) And in this particular case, I consider `datetime` use to be redundant (my version of tests takes less time on my PC).